Implementado boas praticas no retorno da partida

lusoal opened the pull request:
- Adicionado endpoint para retornar as partidas que estao ocorrendo.
- Adicionado tratamento de resposta de erro com ExceptionController.

gb-lucas-duarte committed:
Implementado boas praticas no retorno da partida, contruido objeto e implementado metodo para tratar Exception

gb-lucas-duarte committed:
Variaveis de ambiente no application.properties

LeoAraujo97 commented on src/main/java/lucas/duarte/jazz/controller/PartidaController.java:
```diff
@@ -20,34 +24,64 @@
+	public ResponseEntity<?> listAllpartidas() {
+		List<Partida> partidas = partidaServ.getAllPartidas();
+
+		if (partidas.isEmpty()) {
```
essa validação pode estar na service, e retorna o Response de lá caso não exista partidas.

lusoal replied:
Nao, como disse no grupo do wpp, o Response Entity tem que ser tratato no controller somente, e não na camada de regra de negocio, isso está nas boas práticas do Clean Architecture, pois assim diminuimos o acoplamento do controller com o service, vamos refatorar todas.

lusoal replied:
na verdade a validação do is empty é equivalente a if null no controller, por tanto não tem diferença nesse caso, apenas retorna um Json vazio

LeoAraujo97 commented on src/main/java/lucas/duarte/jazz/controller/PartidaController.java:
```diff
@@ -20,34 +24,64 @@
-		System.out.println("Vou cadastrar uma partida");
-		// Validar se partida Ja Existe
-		partidaServ.cadastrarPartida(partida);
+		try {
```
Try Catch não é muito legal de deixar no código, parece que tem falta de validação por parte de quem construiu, além disso essa tentatida de cadastro deve estar na service.

lusoal replied:
Concordo que o Try catch deveria estar no service, vou ajustar.

LeoAraujo97 commented on src/main/java/lucas/duarte/jazz/controller/PartidaController.java:
```diff
@@ -20,34 +24,64 @@
-		// Retorna objeto ja instanciado
 		Partida partida = partidaServ.getpartidaById(id);
-		// Nao achou a partida
 		if (partida == null) {
```
validação na service

lusoal replied:
a validação já ocorre no service, porém aqui estou verificando se existe partida, por isso valido se é null ou não.

LeoAraujo97 commented on src/main/java/lucas/duarte/jazz/controller/PartidaController.java:
```diff
@@ -20,34 +24,64 @@
+
+		List<Partida> partidasAndamento = partidaServ.getPartidaEmAndamento();
+
+		if (partidasAndamento.isEmpty()) {
```
service também.

LeoAraujo97 reviewed (changes requested):
Lucão, alguns pontos que eu acho que devem ser melhorados.

gb-lucas-duarte committed:
Ajustado try except nas partidas e melhorado o response entity return

gb-lucas-duarte committed:
Adicionado iniciar e finalizar partidas.

- Finalizada feture de inicar partida.
- Finalizada fetura de finalizar partida.
- Implementado retorno dos campeonatos com o nome e id utilizando JSON.

gb-lucas-duarte committed:
Refatorado Dockerfile para utilizar multistaging e alterado porta da aplicação
